src/Text/Libyaml.hs

```diff
@@ -77,6 +79,12 @@ data Style = Any
            | PlainNoTag
     deriving (Show, Read, Eq, Enum, Bounded, Ord, Data, Typeable)
 
+data SequenceStyle = AnySequence | BlockSequence | FlowSequence
```

Comment:
Is there a case where `AnySequence` will come back from the parser, or is only useful for the emitter?

Reply:
AIUI from a reading of the libyaml source, AnySequence and AnyMapping are never produced by the parser (parser.c only references the flow and block constants). They correspond directly to the C enums and the Anys just express no preference as you'd expect. In reality, block is the default barring a few conditions:

e.g.
```c
    if (emitter->flow_level || emitter->canonical
            || event->data.mapping_start.style == YAML_FLOW_MAPPING_STYLE
            || yaml_emitter_check_empty_mapping(emitter)) {
        emitter->state = YAML_EMIT_FLOW_MAPPING_FIRST_KEY_STATE;
    }
    else {
        emitter->state = YAML_EMIT_BLOCK_MAPPING_FIRST_KEY_STATE;
    }
```

(I admit to some unease about the naming I've chosen here, particularly given this is an API change. Happy to accept alternative suggestions!)